packages/pwa-kit-cli/src/ssr/server/build-dev-server.js

```diff
@@ -149,13 +147,8 @@ export const DevServerMixin = {
             })
         })
 
-        app.use('/', (req, res, next) => {
-            if (app.__webpackReady()) {
-                middleware(req, res, next)
-            } else {
-                res.redirect(301, '/__mrt/loading-screen/index.html?loading=1')
-            }
-        })
+        const middleware = webpackHotServerMiddleware(app.__compiler)
+        this._useWebpackHotServerMiddleware(app, middleware)
```

Comment:
A bit of refactoring, so that it can be called for testing purpose.

Reply:
I'm guessing the reason for refactoring is so you don't have use some kind of module mock for the webpackHotServerMiddleware correct?

I think your approach is good but `_useWebpackHotServerMiddleware` isn't really representative of what is actually happening in that function. I believe if we were to come up with a name it would be something along the lines of `useLoadingPageMiddleware` because what is being passed to `app.use` is in fact a middle ware definition which in turn uses another middleware (WebpackHotServerMiddleware).

We could think of this as a naming issue, but I think there is a simpler solution and that would be to create a function called `_getHotServerMiddleware(app)` (public or private, i'm not sure). It might look like this:

```
getHotServerMiddleware(app): {
    return webpackHotServerMiddleware(app.__compiler)
}
```

You'd use it in your app, replacing: 

```
const middleware = webpackHotServerMiddleware(app.__compiler)
```

with 

```
const middleware = getHotServerMiddleware(app)
```

Now inside your test you can define that function on the `NoWebpackDevServerFactory` like:

```
const NoWebpackDevServerFactory = {
    ...DevServerFactory,
    ...{
        addSSRRenderer() {},
        addSDKInternalHandlers() {},
        getRequestProcessor() {},
        getHotServerMiddleware() {} <-- new
    }
}
```